ara/exec/parser/manifest_parser.h

```diff
@@ -11,27 +12,74 @@ struct ExecutionManifest
 {
     // TODO add proper members
 
-    bool operator==(const ExecutionManifest&) const
+    bool operator==(const ExecutionManifest&) const noexcept
     {
         return true;
     }
+    bool operator!=(const ExecutionManifest& other) const noexcept
+    {
+        return !(*this == other);
+    }
 };
 
 struct MachineManifest
 {
     // TODO add proper members
 
-    bool operator==(const MachineManifest&) const
+    bool operator==(const MachineManifest&) const noexcept
     {
         return true;
     }
+    bool operator!=(const MachineManifest& other) const noexcept
+    {
+        return !(*this == other);
+    }
+};
+
+class Manifest  // Simple union wrapper- no std::variant in c++14
+{
+public:
+    enum class Type { kExecution, kMachine, kUnknown };
+
+public:
+    Manifest(){};
+    Type get_type() const
+    {
+        return type_;
+    }
+    ExecutionManifest get_execution() const
+    {
+        return exec_man;
+    }
+    MachineManifest get_machine() const
+    {
+        return mach_man;
+    }
+
+    void set_execution(const ExecutionManifest& man)
+    {
+        type_ = Type::kExecution;
+        exec_man = man;
+    }
+    void set_machine(const MachineManifest& man)
+    {
+        type_ = Type::kMachine;
+        mach_man = man;
+    }
+
+private:
+    Type type_{Type::kUnknown};
+    union
+    {
+        ExecutionManifest exec_man;
+        MachineManifest mach_man;
+    };
 };
 
-ExecutionManifest parse_execution_manifest(const std::string& /*path*/);
-MachineManifest parse_machine_manifest(const std::string& /*path*/);
+Manifest parse_manifest(const std::string& /*path*/) noexcept(false);
```

Comment:
What is the purpose of `noexcept(false)`? Is there any overload that does not throw?

Reply:
It's only for notation/readability purpose- `noexcept(false)` explicitly indicates that this function may throw. I've seen this in many places in AUTOSAR Guidelines and I just liked that approach